Adds Peaches and a Mutation Called Impeach

SushiFish100 opened the pull request:
## About The Pull Request

Credit to Cobby for coming up with the idea which was too good to pass up.

**Peaches:**
![peach_fruit](https://user-images.githubusercontent.com/26680809/122353093-b86bb600-cf1d-11eb-9eaf-63a906b5e58e.png)![peach_seed](https://user-images.githubusercontent.com/26680809/122353101-ba357980-cf1d-11eb-9a81-224fcd47e0c7.png)![peach](https://user-images.githubusercontent.com/26680809/122351536-36c75880-cf1c-11eb-8630-31300d28189e.png)
- Seeds an be accessed via the MegaSeed Vendor, when mutated they turn into impeaches.
- Can be juiced to produce peach juice, which are used in bar drinks that are already in the game. 
- They are a great new food to enjoy while everything around you is crumbling.

**Impeaches:**
![impeach_fruit](https://user-images.githubusercontent.com/26680809/122353109-bbff3d00-cf1d-11eb-800a-8ddeb774eb59.png)![impeach_seed](https://user-images.githubusercontent.com/26680809/122353118-be619700-cf1d-11eb-855d-28172b8e7226.png)![impeach](https://user-images.githubusercontent.com/26680809/122351486-2b742d00-cf1c-11eb-9570-c7ef8eb7fe63.png)
- Impeaches are basically locker grenades to impeach that pesky Captain you didn't vote for. 
- Give the impeach a good squeeze in your hands and get ready to throw that sucker before you put yourself inside a giant peach locker. 
- The person that threw the impeach is the only person that can lock and unlock the giant peach locker that is created from it.

**Peach Locker:**  Vive la révolution!
![plocker](https://user-images.githubusercontent.com/26680809/122353125-bf92c400-cf1d-11eb-87ee-1ea0ef006b9f.png)![plocker2](https://user-images.githubusercontent.com/26680809/122353128-c0c3f100-cf1d-11eb-9d05-a3f3f82c8830.png)![degrading](https://user-images.githubusercontent.com/26680809/122353131-c15c8780-cf1d-11eb-821e-5b41b3c03629.png)![degrading2](https://user-images.githubusercontent.com/26680809/122353138-c3264b00-cf1d-11eb-8f78-39faf83bad27.png)
- Peach lockers are a quick and easy way to slow down your foes or to mass impeach heads that are getting out of control. 
- The locker slowly degrades over time as the local rats and insects get to it. 
- Can only be locked and unlocked by the person who threw the impeach.
- Stats heavily rely on how good the impeaches potency and endurance are.
- Cannot be emagged or emped to get the locker open.

Other Icons:
![a1](https://user-images.githubusercontent.com/26680809/122350650-6590ff00-cf1b-11eb-81b5-68226fb0f097.png)![a2](https://user-images.githubusercontent.com/26680809/122350665-69248600-cf1b-11eb-938b-f48e14d2c198.png)![a3](https://user-images.githubusercontent.com/26680809/122350671-6b86e000-cf1b-11eb-8528-22ccd02b6f5a.png)![a4](https://user-images.githubusercontent.com/26680809/122350683-6f1a6700-cf1b-11eb-954a-d84150f4ef10.png)![a5](https://user-images.githubusercontent.com/26680809/122350691-70e42a80-cf1b-11eb-901e-dd48e64e78a6.png)![a6](https://user-images.githubusercontent.com/26680809/122350696-72adee00-cf1b-11eb-94c7-d7df86bd81b1.png)![dead](https://user-images.githubusercontent.com/26680809/122351552-39c24900-cf1c-11eb-8926-0085ac5b8697.png)

## Why It's Good For The Game

The impeach locker adds a really fun new element to the game for everyone, be it pranks or a new tool for really robust people.

## Changelog
:cl:
add: Added peaches to botany and a mutation called impeach which is a locker grenade
add: Added peach lockers created by impeach grenades
imageadd: Added peach and impeach seed, fruit and growing icons
imageadd: Added peach locker icons
soundadd: Added a twisting sound effect
/:cl:

SushiFish100 committed:
Added Peaches and a Impeach Mutation

SushiFish100 committed:
Added Peach Lockers (Plockers)

- Created when impeaches are activated.
- Very weak considering it is a plant.
- Degrades and eventually destroys itself over time.
- Can only be locked and unlocked by the person who activated and threw the impeach.
- Cannot be emagged.
- Cannot be emped.

SushiFish100 committed:
Added Peach Seeds to MegaSeed Vendor

SushiFish100 committed:
Added Peach Seeds to MegaSeed Ventor

SushiFish100 committed:
Added Twisting Sound Effect

- Used when an impeach is activated.

SushiFish100 committed:
Added Peach and Impeach Seed Icons

SushiFish100 committed:
Added Peach and Impeach Fruit Icons

SushiFish100 committed:
Added Peach and Impeach Growing Icons

SushiFish100 committed:
Added Peach Locker and Degrading Plocker Icons

SushiFish100 committed:
Updated DME

maxymax13 commented:
> 
> 
> Effectively stunning people for between 10 seconds and 12 minutes with a single hit from a meme item that can be obtained by non-antagonists at near-roundstart is fair and balanced.

so is the rest of botany 😉

Krysonism reviewed (changes requested):
The harvested peach and the locker do not have the required sprite quality.

Our sprites should generally have an outline unless there is a compelling reason to omit one.

The shading is also not quite there, for example, it looks like there is a weird indentation in the lower right.

The locker will also need an outline.

The growing plant sprites and the seed packet sprites are fine.

I have some experience spriting plants and I am willing to help you out with the peach sprite if you wish.

SushiFish100 commented:
> The harvested peach and the locker do not have the required sprite quality.
> 
> Our sprites should generally have an outline unless there is a compelling reason to omit one.
> 
> The shading is also not quite there, for example, it looks like there is a weird indentation in the lower right.
> 
> The locker will also need an outline.
> 
> The growing plant sprites and the seed packet sprites are fine.
> 
> I have some experience spriting plants and I am willing to help you out with the peach sprite if you wish.

That'd be great if you wanted to do that for me. I'll also try and find someone to help me out with the locker icons. Thanks for your help by the way, really appreciate it.

ATH1909 commented:
le crew-obtainable locker staff has arrived

ATH1909 commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,90 @@
 	new /obj/item/secateurs(src)
+
+//Plocker
+/obj/structure/closet/secure_closet/plocker
```
making this a subtype of /obj/structure/closet/decay could remove the need for some redundant code

ATH1909 commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,90 @@
+//Plocker
+/obj/structure/closet/secure_closet/plocker
+	name = "peach locker"
+	desc = "A huge hollow pitless peach."
```
```suggestion
	desc = "A huge, hollow, pitless peach."
```

ATH1909 commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,90 @@
+							"<span class='notice'>You [locked ? null : "un"]lock [src].</span>")
+			update_appearance()
+		else if(!silent)
+			to_chat(user, "<span class='alert'>Access Denied.</span>")
```
why is a peach giving you an "Access Denied." error?

ATH1909 commented:
imo, it'd be best if anyone other than the person inside of the peach could open the peach, so that people inside of the peach can beg passerby and/or their allies to let them out.

that'd make the stats of the peach matter less, though- maybe you could let the person trapped inside of the peach attack it from the inside?

ATH1909 commented on code/modules/hydroponics/grown/peach.dm:
```diff
@@ -0,0 +1,97 @@
+/obj/item/food/grown/peach/impeach
+	seed = /obj/item/seeds/peach/impeach
+	name = "impeach"
+	desc = "A quick way to impeach your pets, friends or even your whole family!"
```
```suggestion
	desc = "A quick way to impeach your pets, your friends, or even your whole family!"
```

ExcessiveUseOfCobblestone replied:
oxford comma...

ATH1909 commented:
also, it might be humorous if impeaches could be distilled into one of those law-related drinks- maybe beepsky smash or triple sec?

MrMelbert commented on code/modules/hydroponics/grown/peach.dm:
```diff
@@ -0,0 +1,97 @@
+/**
+* Creates and spawns a giant peach locker.
+*/
+/obj/item/food/grown/peach/impeach/proc/create_plocker(user)
+	var /obj/structure/closet/secure_closet/plocker/P = new /obj/structure/closet/secure_closet/plocker(get_turf(src.loc))
+	P.visible_message("<span class='notice'>[src] suddenly turns into a giant peach!</span>")
+	P.obj_integrity = round(P.max_integrity + (seed.potency / 4))
+	P.damage_deflection = round(seed.endurance / 15)
+	P.breakout_time += round(seed.endurance / 2)
+	P.lifespan += round(seed.lifespan * 6)
+	P.owner = user
+	playsound(loc, 'sound/misc/moist_impact.ogg', 200, TRUE)
+	return P
```
use better variable names (maybe `spawned_locker`) instead of `P`. Same goes for all the rest of the code

SushiFish100 replied:
True, P is a pretty bad name. Renamed the variable to spawned_locker instead.

MrMelbert commented on code/modules/hydroponics/grown/peach.dm:
```diff
@@ -0,0 +1,97 @@
+/**
+* Activates the timer for the impeach
+*/
+/obj/item/food/grown/peach/impeach/attack_self(mob/user)
+	icon_state = "impeach_active"
+	activated = TRUE
+	addtimer(CALLBACK(src, .proc/dry_activate, user), 3 SECONDS)
+	playsound(loc, 'sound/misc/twisting.ogg', 60, TRUE)
+
+/**
+* When coming in contact with a mob while active it will put them into a locker
+*/
+/obj/item/food/grown/peach/impeach/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	. = ..()
+	if(QDELETED(src) || activated == FALSE)
+		return
+	var /obj/structure/closet/secure_closet/plocker/P = create_plocker(throwingdatum.thrower)
+	if(P.insert(hit_atom)) //when the impeach hits a mob
+		log_combat(throwingdatum.thrower, hit_atom, "trapped", /obj/item/food/grown/peach/impeach)
+	else //when the impeach hits anything that isnt a mob
+		for(var/mob/M in get_turf(P))
+			if(P.insert(M))
+				log_combat(throwingdatum.thrower, M, "trapped", /obj/item/food/grown/peach/impeach)
+	qdel(src)
```
I suggest turning this behavior into a plant gene datum instead of having it on the plant itself, and making it incompatible with with other `ATTACK_SELF_ID` and `THROW_IMPACT_ID` plant genes, like liquid contents. See killer tomatos, for example.

SushiFish100 replied:
Sounds good, I'll look into it right now

SushiFish100 committed:
Better Variable Names

SushiFish100 committed:
Updated Peach Vars

Fikou commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,90 @@
+	var/owner = null
+	var/lifespan = 1800
+
+/**
+* Overrided to include a different sound when attacked
+*/
```
procs that already exist shouldnt have autodoc comments

Fikou commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,90 @@
+	close_sound = "sound/misc/moist_impact.ogg"
+	open_sound = "sound/misc/soggy.ogg"
+	breakout_time = 100
+	var/isDegrading = FALSE
+	var/owner = null
+	var/lifespan = 1800
```
while these vars youre just now making should

SushiFish100 replied:
Alright, I got rid of bunch of doc comments that weren't needed and added comments to new procs and variables only.

Fikou commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,90 @@
 	new /obj/item/secateurs(src)
+
+//Plocker
+/obj/structure/closet/secure_closet/plocker
```
plocker is a confusing path if anyone wanted to spawn it, why not just secure_closet/peach

SushiFish100 replied:
True, renamed plocker obj path to peach.

coiax commented:
@ExcessiveUseOfCobblestone Correct me if I'm wrong, but aren't bar drinks involving peach juice "balanced" around the relative rarity of maintenance peaches?

SushiFish100 commented:
> 
> 
> @ExcessiveUseOfCobblestone Correct me if I'm wrong, but aren't bar drinks involving peach juice "balanced" around the relative rarity of maintenance peaches?

I'm going to make a change in that instead of being available through the MegaSeed vendor, there will be chance that canned peaches will be a whole peach instead.

SushiFish100 committed:
Updated Peach Lockers

- Removed unneeded comments.
- Added comments to new vars and procs only.
- Renamed the locker object path from "plocker" to "peach".

SushiFish100 committed:
Updated Peaches and Impeaches

- Removed unneeded comments.
- Added comments for new vars and procs only.
- Created better names for variables.

optimumtact commented:
this is just wacky I don't like it :-1:

ilikesaying commented:
oranges hates peaches confirmed

YakumoChen commented:
## About the Pull Request 

Adds CuffLynx, a research able nonlethal security gun printed fully from a seclathe
Features:
-shoots handcuffs that instantly subdue criminals on hit
-100 shot capacity. Fits on the belt! 
-Adjustable resist time between 10 seconds and 10 minutes (let's face it, it's always 10 minutes)
-MagnumSteel alloy resists freedom implants
-biometric security features ensure nobody else can steal your gun or free captured CRIMINAL SCUM

## Why it's goodfor the game

I made a funny pun with the name guys

carshalash commented:
Needs a high rate of bug infestation

legoscape commented:
Wondering If it should be possible to eat the locker?

the-orange-cow commented:
> Wondering If it should be possible to eat the locker?

Eating your way out should be the default way of escaping.

Krysonism commented:
> 
> 
> > Wondering If it should be possible to eat the locker?
> 
> Eating your way out should be the default way of escaping.

This is very possible to implement with the edible component.

TheVekter commented:
The amount you have to eat to get out should scale with stats.

BebeYoshi commented:
I think making a special IMMUTABLE gene introduced in #59107 to the Impeachment plant would be good, and it would be amazing/funny if it made the plant backfire if you do not hold it with gloves (make it so it locks you in the peach locker but does not bound anyone as the user). 
Otherwise it looks cool despite the balance issues as you can effectively locker someone forever if no one helps them, or just you know lock everyone you see.

interestingusernam3 commented:
Check icon diffs, you seem to have made some accidental alterations to several files.

evsey9 commented:
maybe the impeach should trap someone if you forcefeed it to them, instead of throwing it at them? That way, it could still be effective/useful if you manage to stun someone, and hopefully a bit more balanced

ExcessiveUseOfCobblestone commented:
> @ExcessiveUseOfCobblestone Correct me if I'm wrong, but aren't bar drinks involving peach juice "balanced" around the relative rarity of maintenance peaches?

i have no idea wbout the balance of maint peaches and offshoots.

also lmfao

ExcessiveUseOfCobblestone commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,75 @@
+	close_sound = "sound/misc/moist_impact.ogg"
+	open_sound = "sound/misc/soggy.ogg"
+	breakout_time = 100
+	var/isDegrading = FALSE //Used to check if the peach locker is degrading
+	var/owner = null //The owner of the locker which is set when an impeach is activated
+	var/lifespan = 3 MINUTES //Right when the peach locker is created it has this amount of time before it starts degrading
```
You have generated new variables/procs/defines without proper documentation for the autodoc to work.

Please click [here ](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)for a guide on how to properly document these items.

ExcessiveUseOfCobblestone commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,75 @@
+			else
+				playsound(src, 'sound/weapons/tap.ogg', 50, TRUE)
+		if(BURN)
+			playsound(src.loc, 'sound/items/welder.ogg', 100, TRUE)
```
```suggestion
			playsound(loc, 'sound/items/welder.ogg', 100, TRUE)
```

ExcessiveUseOfCobblestone commented on code/game/objects/structures/crates_lockers/closets/secure/hydroponics.dm:
```diff
@@ -11,3 +11,75 @@
+/**
+* Begins the process for the locker to degrade
+*/
+/obj/structure/closet/secure_closet/peach/proc/start_degrading()
+	isDegrading = TRUE
+	icon_state = "plocker_degrade"
+	desc += " There are ants all over it!"
+	update_appearance()
+	degrade()
+
+/**
+* Degrades the peach locker over time
+*/
+/obj/structure/closet/secure_closet/peach/proc/degrade()
+	if(QDELETED(src) || isDegrading == FALSE)
+		return
+	src.take_damage(10,BRUTE,"",FALSE)
+	if(obj_integrity > 0)
+		addtimer(CALLBACK(src, .proc/degrade), 1 MINUTES)
```
use proper processing.

https://github.com/tgstation/tgstation/blob/55b92786dcf0f9c5172efd111a6ecbf83ec0dae2/code/game/objects/items/tools/weldingtool.dm

for example (search process and/or processing)

ExcessiveUseOfCobblestone replied:
You have generated new variables/procs/defines without proper documentation for the autodoc to work.

Please click [here ](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)for a guide on how to properly document these items.

ExcessiveUseOfCobblestone commented on code/modules/hydroponics/grown/peach.dm:
```diff
@@ -0,0 +1,91 @@
+	name = "impeach"
+	desc = "A quick way to impeach your pets, friends or even your whole family!"
+	icon_state = "impeach"
+	var/activated = FALSE //Has the impeach been activated yet?
```
You have generated new variables/procs/defines without proper documentation for the autodoc to work.

Please click [here ](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)for a guide on how to properly document these items.

ExcessiveUseOfCobblestone commented on code/modules/hydroponics/grown/peach.dm:
```diff
@@ -0,0 +1,91 @@
+* Creates and spawns a giant peach locker.
+*/
+/obj/item/food/grown/peach/impeach/proc/create_locker(user)
+	var /obj/structure/closet/secure_closet/peach/spawned_locker = new /obj/structure/closet/secure_closet/peach(get_turf(src.loc))
```
```suggestion
	var /obj/structure/closet/secure_closet/peach/spawned_locker = new /obj/structure/closet/secure_closet/peach(get_turf(src))
```

ExcessiveUseOfCobblestone commented:
this really should be capped at like 1 per harvest, the excuse being the weight of multiple impeaches would harm the tree.
